Svc/FpySequencer/FpySequencerValidationState.cpp

```diff
@@ -0,0 +1,181 @@
+    // FpySequencer. this is because you must have a buffer big enough to fit the
+    // header of a sequence
+    FW_ASSERT(bytes >= Fpy::Header::SERIALIZED_SIZE, static_cast<FwAssertArgType>(bytes));
+    FwSizeType originalBytes = bytes;
+    bool recoverable = false;
+    this->m_allocatorId = identifier;
+    U8* allocatedMemory = static_cast<U8*>(allocator.allocate(identifier, bytes, recoverable));
+    // if this fails, unable to allocate the requested amount of money
+    FW_ASSERT(bytes >= originalBytes, static_cast<FwAssertArgType>(bytes));
+    this->m_sequenceBuffer.setExtBuffer(allocatedMemory, bytes);
+}
+
+void FpySequencer::deallocateBuffer(Fw::MemAllocator& allocator) {
+    allocator.deallocate(this->m_allocatorId, this->m_sequenceBuffer.getBuffAddr());
+    this->m_sequenceBuffer.clear();
+}
+
+void FpySequencer::updateComputedCRC(const U8* buffer, FwSizeType bufferSize) {
+    FW_ASSERT(buffer);
+    for (FwSizeType index = 0; index < bufferSize; index++) {
+        this->m_computedCRC = static_cast<U32>(update_crc_32(this->m_computedCRC, static_cast<char>(buffer[index])));
+    }
+}
+
+// loads the sequence in memory, and does header/crc/integrity checks.
+// return true if sequence is valid
+bool FpySequencer::validate() {
+    FW_ASSERT(m_sequenceFilePath.length() > 0);
+
+    // crc needs to be initialized with a particular value
+    // for the calculation to work
+    this->m_computedCRC = CRC_INITIAL_VALUE;
+
+    Os::File sequenceFile;
+    Os::File::Status openStatus = sequenceFile.open(m_sequenceFilePath.toChar(), Os::File::OPEN_READ);
+
+    if (openStatus != Os::File::Status::OP_OK) {
+        this->log_WARNING_HI_FileOpenError(m_sequenceFilePath, static_cast<I32>(openStatus));
+        return false;
+    }
+
+    // read header bytes
+    bool readStatus = readBytes(sequenceFile, Fpy::Header::SERIALIZED_SIZE);
+    if (!readStatus) {
+        return false;
+    }
+
+    // deser header
+    Fw::SerializeStatus deserStatus = m_sequenceBuffer.deserialize(m_sequenceObj.getheader());
+    if (deserStatus != Fw::SerializeStatus::FW_SERIALIZE_OK) {
+        this->log_WARNING_HI_FileReadDeserializeError(m_sequenceFilePath, static_cast<I32>(deserStatus),
+                                                      m_sequenceBuffer.getBuffLeft(), m_sequenceBuffer.getBuffLength());
+        return false;
+    }
+
+    // check matching schema version
+    if (m_sequenceObj.getheader().getschemaVersion() != Fpy::SCHEMA_VERSION) {
+        this->log_WARNING_LO_WrongSchemaVersion(Fpy::SCHEMA_VERSION, m_sequenceObj.getheader().getschemaVersion());
+        return false;
+    }
+
+    if (m_sequenceObj.getheader().getargumentCount() > Fpy::MAX_SEQUENCE_ARG_COUNT) {
+        this->log_WARNING_HI_TooManyArgs(m_sequenceObj.getheader().getargumentCount(), Fpy::MAX_SEQUENCE_ARG_COUNT);
+        return false;
+    }
+
+    if (m_sequenceObj.getheader().getstatementCount() > Fpy::MAX_SEQUENCE_STATEMENT_COUNT) {
```

Comment:
static assert MAX_SEQUENCE_STATEMENT_COUNT <= max U16 otherwise there is another implicit statement count.

Use aliases when available.